can/listener.py

```diff
@@ -49,25 +71,48 @@ class BufferedReader(Listener):
     A BufferedReader is a subclass of :class:`~can.Listener` which implements a
     **message buffer**: that is, when the :class:`can.BufferedReader` instance is
     notified of a new message it pushes it into a queue of messages waiting to
-    be serviced.
+    be serviced. The messages can then be fetched with
+    :meth:`~can.BufferedReader.get_message`.
+
+    Putting in messages after :meth:`~can.BufferedReader.stop` has be called will raise
+    an exception, see :meth:`~can.BufferedReader.on_message_received`.
+
+    :attr bool is_stopped: ``True`` iff the reader has been stopped
     """
 
     def __init__(self):
-        self.buffer = queue.Queue(0)
+        # 0 is "infinite" size
+        self.buffer = SimpleQueue(0)
+        self.is_stopped = False
 
     def on_message_received(self, msg):
-        self.buffer.put(msg)
+        """Append a message to the buffer.
+
+        :raises: BufferError
+            if the reader has already been stopped
+        """
+        if self.is_stopped:
+            raise BufferError("reader has already been stopped")
```

Comment:
Not sure BufferError is appropriate here. Maybe just RuntimeError or something.

Reply:
> exception BufferError: Raised when a buffer related operation cannot be performed.

It sort of fits, but RuntimeError does as well. I'll use RuntimeError.